server/src/Api/Controllers/CoursesController.cs

```diff
@@ -44,15 +46,15 @@ public async Task<IActionResult> GetAllCourses()
     public async Task<IActionResult> UpdateCourse(int courseId, [FromBody] UpdateCourseDto request)
     {
         request.ThrowIfRequestArgumentNull(nameof(request));
-        var isSuccess = await _courseService.UpdateCourse(courseId, request);
-        return isSuccess ? Ok() : BadRequest();
+        var result = await _courseService.UpdateCourse(courseId, request).ToActionResult();
+        return result;
     }
 
     [HttpDelete]
     [Route("{courseId}")]
     public async Task<IActionResult> DeleteCourse(int courseId)
     {
-        var isSuccess = await _courseService.DeleteCourse(courseId);
-        return isSuccess ? Ok() : NotFound();
+        var result = await _courseService.DeleteCourse(courseId).ToActionResult();
```

Comment:
```suggestion
        return await _courseService.DeleteCourse(courseId).ToActionResult();
```

here and in other methods